e2e/challenge-title.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { test, expect } from '@playwright/test';
+import translations from '../client/i18n/locales/english/translations.json';
+
+test.beforeEach(async ({ page }) => {
+  await page.goto(
+    '/learn/information-security/python-for-penetration-testing/developing-a-port-scanner'
+  );
+});
+
+test.afterEach(async ({ page }) => {
+  await page.close();
+});
+
+test.describe('Challenge Title Component (signed out)', () => {
+  test('should render correctly', async ({ page }) => {
+    await expect(page.getByLabel('Passed')).not.toBeVisible();
+
+    await expect(page.getByText('Developing a Port Scanner')).toBeVisible();
+  });
+
+  test('should not display GreenPass after challenge completion', async ({
+    page
+  }) => {
+    await page
+      .getByText(
+        'If there is an error or if no host is found, .connect() raises an exception while .connect_ex() returns an error code.'
+      )
+      .click();
+
+    await page
+      .getByRole('button', { name: translations.buttons['check-answer'] })
+      .click();
+
+    await page
+      .getByRole('button', { name: translations.buttons['go-to-next'] })
+      .click();
+
+    await page.waitForLoadState('domcontentloaded');
+
+    // After clicking 'go-to-next' button, page redirects to courses list.
+    // Returning back to the challenge, to verify that GreenPass is rendered on the challenge itself.
+    await page.goto(
+      '/learn/information-security/python-for-penetration-testing/developing-a-port-scanner'
+    );
```

Comment:
The `goto` raises an `NS_BINDING_ABORTED` error (or a navigation interruption error) for Firefox, Webkit, and Mobile Safari. The `waitForLoadState` was an attempt at resolving these errors for the browsers mentioned, but the problem still persists.